topydo/ui/TodoListWidget.py

```diff
@@ -101,132 +119,199 @@ def _scroll_to_bottom(self, p_size):
                 self.keystate = None
-
-            return
-        elif self.keystate == 'r':
-            if p_key.isalpha():
-                self._pri_selected_item(p_key)
-            self.keystate = None
             return
-
-        if p_key == 'x':
-            self._complete_selected_item()
-        elif p_key == 'p':
-            self.keystate = 'p'
-        elif p_key == 'd':
-            self._remove_selected_item()
-        elif p_key == 'e':
-            self._edit_selected_item()
-            # force screen redraw after editing
-            urwid.emit_signal(self, 'refresh')
-        elif p_key == 'r':
-            self.keystate = 'r'
-        elif p_key == 'u':
-            urwid.emit_signal(self, 'execute_command', "revert")
-        elif p_key == 'j':
-            self.listbox.keypress(p_size, 'down')
-        elif p_key == 'k':
-            self.listbox.keypress(p_size, 'up')
-        elif p_key == 'home':
-            self._scroll_to_top(p_size)
-        elif p_key == 'G' or p_key == 'end':
-            self._scroll_to_bottom(p_size)
-        elif p_key == 'g':
-            self.keystate = 'g'
         else:
-            return self.listbox.keypress(p_size, p_key)
+            if shortcut in keystates:
+                self.keystate = shortcut
+            else:
+                try:
+                    # Check whether current keystate matches built-in 'postpone'
+                    # action.
+                    pp_mode = keymap[self.keystate]
+                    if pp_mode in ['postpone', 'postpone_s']:
+                        if self._postpone_selected(p_key, pp_mode) is not None:
+                            self.keystate = None
+                    return
+                except KeyError:
+                    if not self.keystate:
+                        # Single key that is not described in keymap config.
+                        return self.listbox.keypress(p_size, p_key)
+                    self.keystate = None
+            return
 
     def mouse_event(self, p_size, p_event, p_button, p_column, p_row, p_focus):
         if p_event == 'mouse press':
             if p_button == 4:  # up
                 self.listbox.keypress(p_size, 'up')
                 return
-            elif p_button == 5:  #down:
+            elif p_button == 5:  # down:
                 self.listbox.keypress(p_size, 'down')
                 return
 
-        return super().mouse_event(p_size, p_event, p_button, p_column, p_row, p_focus)  # pylint: disable=E1102
+        return super().mouse_event(p_size,  # pylint: disable=E1102
+                                   p_event,
+                                   p_button,
+                                   p_column,
+                                   p_row,
+                                   p_focus)
 
     def selectable(self):
         return True
 
-    def _command_on_selected(self, p_cmd_str):
+    def _command_on_selected(self, p_cmd_str, p_execute_signal):
```

Comment:
For this method and the methods below: method names should be a verb of what they do. This helps readability, it was not obvious to me what these methods _do_.